tests/ui/closures/issue-84128.stderr

```diff
@@ -6,6 +6,13 @@ LL |         Foo(())
    |         |
    |         arguments to this struct are incorrect
    |
+help: the type constructed contains `()` due to the type of the argument passed
+  --> $DIR/issue-84128.rs:13:9
+   |
+LL |         Foo(())
+   |         ^^^^--^
+   |             |
+   |             this argument influences the type of `Foo`
```

Comment:
I'm noticing that there are some cases where the existing machinery to detect the source of expectations isn't as robust as we'd like, but that's independent of this PR.

Reply:
Filed https://github.com/rust-lang/rust/issues/106821